mcpserver/tools/channels.go

```diff
@@ -254,44 +267,99 @@ func (p *MattermostToolProvider) toolGetChannelInfo(mcpContext *MCPToolContext,
+			}
+			// Real error (network, auth, etc.)
+			return "failed to fetch channel", fmt.Errorf("error fetching channel by ID: %w", err)
 		}
+		channels = []*model.Channel{channel}
 	case args.ChannelDisplayName != "" || args.ChannelName != "":
 		// Prioritize display name over name - try display name first if provided
 		if args.ChannelDisplayName != "" {
-			channel, lastError = p.tryFindChannelByDisplayName(ctx, client, args.ChannelDisplayName, args.TeamID)
-			if channel != nil {
-				break // Found it with display name
+			channels, lastError = p.tryFindChannelByDisplayName(ctx, client, args.ChannelDisplayName, args.TeamID)
+			if lastError != nil {
+				// Real API error (not just zero results)
+				return "failed to search for channels", lastError
+			}
+			if len(channels) > 0 {
+				break // Found results with display name
 			}
 		}
 
-		// If display name didn't work (or wasn't provided), try channel name
+		// If display name didn't find anything (or wasn't provided), try channel name
 		if args.ChannelName != "" {
-			channel, err = p.tryFindChannelByName(ctx, client, args.ChannelName, args.TeamID)
+			channels, err = p.tryFindChannelByName(ctx, client, args.ChannelName, args.TeamID)
 			if err != nil {
-				// If we also failed with display name, return combined error message
-				if lastError != nil {
-					return fmt.Sprintf("channel not found by display name '%s' or name '%s'", args.ChannelDisplayName, args.ChannelName),
-						fmt.Errorf("display name error: %v; name error: %v", lastError, err)
-				}
-				return fmt.Sprintf("channel not found by name '%s'", args.ChannelName), err
+				// Real API error
+				return "failed to search for channels", err
 			}
-		} else if lastError != nil {
-			// Only display name was provided and it failed
-			return fmt.Sprintf("channel not found by display name '%s'", args.ChannelDisplayName), lastError
 		}
 
-		if channel == nil {
-			return "no channel found with the provided parameters", fmt.Errorf("channel lookup failed")
+		// At this point, channels could be empty (no results) but that's not an error
+		if len(channels) == 0 {
+			// Build helpful "not found" message with actionable next steps for LLM
+			var notFoundMsg strings.Builder
+
+			// What was searched
+			switch {
+			case args.ChannelDisplayName != "" && args.ChannelName != "":
+				notFoundMsg.WriteString(fmt.Sprintf("No channels found matching display name '%s' or name '%s'.", args.ChannelDisplayName, args.ChannelName))
+			case args.ChannelDisplayName != "":
+				notFoundMsg.WriteString(fmt.Sprintf("No channels found with display name '%s'.", args.ChannelDisplayName))
+			default:
+				notFoundMsg.WriteString(fmt.Sprintf("No channels found with name '%s'.", args.ChannelName))
+			}
+
+			// Search scope
+			if args.TeamID != "" {
+				notFoundMsg.WriteString(" (searched within specified team)")
+			} else {
```

Comment:
Might be worth including the team name here, or at least the ID since we already have it